lib/pry/repl.rb

```diff
@@ -105,9 +105,16 @@ def read
         indented_val = @indent.indent(val)
 
         if output.tty? && pry.config.correct_indent && Pry::Helpers::BaseHelpers.use_ansi_codes?
+          suffix_whites = if readline_available? && Readline.vi_editing_mode?
+                            _, cols = Terminal.screen_size
+                            cols - indented_val.length
+                          else
+                            false
```

Comment:
I think this value is inconsistent with the other `if` branch, which returns an Integer. I believe it would be simpler to replace this with `[ original_val.length - indented_val.length, 0 ].max` and pass `overhang` on line 117.